dune/stuff/functions/expression.hh

```diff
@@ -202,47 +70,81 @@ public:
     }
   } // ... default_config(...)
 
-  static std::unique_ptr< ThisType > create(const Common::Configuration config = default_config(), const std::string sub_name = static_id())
+  static std::unique_ptr< ThisType > create(const Common::Configuration config = default_config(),
+                                            const std::string sub_name = static_id())
   {
     // get correct config
     const Common::Configuration cfg = config.has_sub(sub_name) ? config.sub(sub_name) : config;
     const Common::Configuration default_cfg = default_config();
+    // get expression
+    ExpressionStringVectorType expression_as_vectors;
+    // try to get expression as FieldVector (if dimRangeCols == 1) or as FieldMatrix (else)
+    try {
+      get_expression_helper(cfg, expression_as_vectors, internal::ChooseVariant< dimRangeCols >());
+    } catch (Exceptions::conversion_error) {
+      // if dimRangeCols == 1 and we could not get expression as FieldVector, get it as FieldMatrix with one col
+      if (dimRangeCols == 1) {
+        get_expression_helper(cfg, expression_as_vectors, internal::ChooseVariant< 2 >());
```

Comment:
That is correct. Actually, anything different from 1 would do the job here. The purpose is to try to get the expression as the first column of a FieldMatrix when getting it as a vector failed before. Maybe that is not the most elegant way to solve it, but the get_expression_helper will get the expression as a FieldMatrix for ChooseVariant< dimRangeCols > with dimRangeCols > 1 and as a FieldVector for dimRangeCols = 1.